src/metabase/upload.clj

```diff
@@ -742,16 +739,15 @@
                                   (remove #(false? (:is_upload %)))
                                   (keep :table_id)
                                   set)
-        uploadable-table-ids (set (uploadable-table-ids table-ids))
-        based-on-upload      (fn [model]
-                               (when-let [dataset_query (:dataset_query model)] ; dataset_query is sometimes null in tests
-                                 (let [query (lib/->pMBQL dataset_query)]
-                                   (when (and (some-> model :query_type name (= "query"))
-                                              (contains? uploadable-table-ids (:table_id model))
-                                              (no-joins? query))
-                                     (lib/source-table-id query)))))]
-    (map #(m/assoc-some % :based_on_upload (based-on-upload %))
-         models)))
+        query?               (fn [model] (= "query" (name (:query_type model "query"))))
```

Comment:
`query` is so overloaded that this name is a little confusing. We frequently refer to `query_type=query` questions as "MBQL questions" so I would change this to `mbql?` instead.
```suggestion
        mbql?               (fn [model] (= "native" (name (:query_type model))))
```
See 
https://github.com/metabase/metabase/blob/1ca43e71eddea3ac8f884a3216fbd7ca03ef4623/src/metabase/mbql/schema.cljc#L1804

Reply:
Cool, I like that predicate name. I think you forgot a `not=` here after changing to `"native"`?

Reply:
Ah yeah sorry, the original implementation was fine. Updated